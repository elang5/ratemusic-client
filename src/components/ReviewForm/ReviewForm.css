.review-form-container {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background-color: rgba(252, 61, 57, 0.73);
}

.review-form {
  text-align: center;
  width: 65%;
  height: 83%;
  border-radius: 5.5px;
  box-shadow: 0 0 3px 2px rgba(0, 0, 0, 0.15);
  background-color: #ffffff;
  color: #1f1f21;
}

.album-title {
  font-size: 24px;
}

.title > label {
  font-size: 18px;
  color: #1f1f21;
}

.title, .rating-select, .content {
  margin: 1rem;
}

.review-form > div > label {
  margin-right: .25rem;
}

#title, #rating, #content{
  margin-top: 1em;
  padding: 15px 18px;
  border-radius: 2.2px;
  border: solid 0.6px rgba(208, 208, 208, 0.5);
  background-color: #e5e9ed;
}

#title, #rating {
  width: 41.5%;
  font-size: 14px;
}

#content {
  max-width: 70%;
  max-height: 160px;
  font-size: 11px;
}

.review-submit-btn {
  width: 41.5%;
  padding: 15px 20px;
  border: none;
  background-color: rgb(252, 60, 57);
  margin: .5em 0;
  border-radius: 2.2px;
  font-size: 20px;
  color: white;
  font-weight: 200;
  cursor: pointer;
}