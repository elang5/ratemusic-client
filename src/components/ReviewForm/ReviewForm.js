import React, { Component } from 'react'
import AlbumApiService from '../../services/albums-api-service'

export class ReviewForm extends Component {
  state = {
    error: null,
    album: [],
    review: [],
    user: []
  }

  componentDidMount() {
    AlbumApiService.getAlbum(this.props.match.params.albumId)
      .then(res => this.setState({ album: res }))
      .catch(err => this.setState({ error: err.error }))
  }

  handleSubmit = e => {
    e.preventDefault()
    const { albumId } = this.props.match.params
    const { title, content, rating } = e.target
    const { history } = this.props
    // can probably clean this up and remove state besides error handling
    AlbumApiService.postReview(albumId, title.value, content.value, rating.value)
      .then(res => console.log(res))
      .then(res => this.setState({ review: res, user: res.user }), history.push(`/albums/${albumId}`))
      .catch(err => this.setState({ error: err.error }))
  }

  render() {
    const { error, album } = this.state
    return (
      <form 
        className="review-form"
        onSubmit={this.handleSubmit}>
          {error && <p>{error}</p>}
          <div className="album-info">
            <h3>{`${album.artist}'s ${album.title}`}</h3>
          </div>
          <div className="title">
            <label htmlFor="title">Title</label>
            <input type="text" id="title"/>
          </div>
          <div className="rating-select">
            <label htmlFor="rating">Rate this album</label>
            <select 
              name="rating" 
              id="rating"
              aria-label='Rate this album'
            >
              <option value="1">1</option>
              <option value="2">2</option>
              <option value="3">3</option>
              <option value="4">4</option>
              <option value="5">5</option>
              <option value="6">6</option>
              <option value="7">7</option>
              <option value="8">8</option>
              <option value="9">9</option>
              <option value="10">10</option>
            </select>
          </div>
          <div className="content">
            <label htmlFor="content">Content</label>
            <textarea name="content" id="content" cols="30" rows="20"></textarea>
          </div>
          <button type="submit">Post review</button>
      </form>
    )
  }
}

export default ReviewForm
