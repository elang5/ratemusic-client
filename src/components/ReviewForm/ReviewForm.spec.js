import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReviewForm from './ReviewForm'

describe(`ReviewForm Component`, () => {

  it(`renders the ReviewForm component`, () => {
    const props = {
      match: {
        params: 1
      }
    }
    const wrapper = shallow(<ReviewForm {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})