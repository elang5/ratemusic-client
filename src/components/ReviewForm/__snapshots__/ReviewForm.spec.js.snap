// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm Component renders the ReviewForm component 1`] = `
<div
  className="review-form-container"
>
  <form
    className="review-form"
    onSubmit={[Function]}
  >
    <div
      className="album-info"
    >
      <h2
        className="album-name"
      >
        undefined
      </h2>
    </div>
    <div
      className="title"
    >
      <label
        htmlFor="title"
      >
        Title: 
      </label>
      <input
        id="title"
        type="text"
      />
    </div>
    <div
      className="rating-select"
    >
      <label
        htmlFor="rating"
      >
        Rating: 
      </label>
      <select
        aria-label="Rate this album"
        id="rating"
        name="rating"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
        <option
          value="8"
        >
          8
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          value="10"
        >
          10
        </option>
      </select>
    </div>
    <div
      className="content"
    >
      <label
        htmlFor="content"
      >
        Content: 
      </label>
      <br />
      <textarea
        cols="30"
        id="content"
        name="content"
        rows="20"
      />
    </div>
    <button
      className="review-submit-btn"
      type="submit"
    >
      Post review
    </button>
  </form>
</div>
`;
